### ISSUE TITLE
curl_reset for PHP5 >= 5.5.0

### ISSUE THREAD

**Issue by interneers** [google/google-api-php-client#793]
Just looking to implement this for verifying In App and Subscriptions, on trying to connect for auth I get the following error:

mod_fcgid: stderr: PHP Fatal error: Call to undefined function GuzzleHttp\\Handler\\curl_reset() in /*mypath*/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php on line 78

Looking in the PHP docs (http://php.net/manual/en/function.curl-reset.php) the function curl_reset is not available below PHP 5.5.0, but the library states PHP 5.4, I have 5.4.45.

The only place I can find it is in the file CurlFactory.php on lines 71 (comment) and 78 (function call). Commenting out the reset has not had any adverse effects, yet!

Wondering what it will be, change to 5.5 or greater or find another way to reset the curl...

Regards,

Jacko

**Comment by bshaffer** [comment]
It sounds like you've managed to install guzzle 6 on Php 5.4, which is not
compatible. Use guzzle 5 instead and you'll be good to go.

**Comment by interneers** [comment]
Brent,

Didn't install with composer but downloaded the 2.0.0RC4 zip file and it seems guzzle 6 is the library that is included in that.

Couldn't see anywhere the option to use an older guzzle using the download method.

Jacko

**Comment by bshaffer** [comment]
You're right. You'll need to use composer to install it instead, for PHP
5.4. Have you tried this? Just download composer (https://getcomposer.org/)
and in your project root run "composer require google/apiclient:~2.0@RC"
and then add "require 'vendor/autoload.php';" in your project.

**Comment by interneers** [comment]
Brent,

Gave it a go, got 5.3 installed on my server but 5.4.45 in the vhost where I was planning to run the service. Could not install from root as it defaulted to 5.3 and failed.

About time I upgraded my server!! On the other hand, I commented out the only curl_reset call and have successfully got my in app product details with no issue or serious fault. Will keep monitoring but will probably upgrade to a younger PHP.

Thanks for the advice.

Jacko
